bin/checks-agent/src/env_provider.rs

```diff
@@ -69,10 +74,24 @@ impl ChecksAgentEnvProvider {
         Ok(Self {
             host_provider,
             autodiscovery_provider,
+            workload_provider: None,
         })
     }
 
     pub fn autodiscovery_provider(&self) -> &BoxedAutodiscoveryProvider {
         &self.autodiscovery_provider
     }
 }
+
+impl EnvironmentProvider for ChecksAgentEnvProvider {
+    type Host = BoxedHostProvider;
+    type Workload = Option<RemoteAgentWorkloadProvider>;
+
+    fn host(&self) -> &Self::Host {
+        &self.host_provider
+    }
+
+    fn workload(&self) -> &Self::Workload {
+        &self.workload_provider
```

Comment:
Just have this return `&None`, since we never actually set `self.workload_provider` to anything other than `None`... and then you can just skip having that field entirely. :D